chain/jsonrpc/jsonrpc-tests/tests/rpc_transactions.rs

```diff
@@ -97,6 +99,43 @@ fn test_send_tx_commit() {
     });
 }
 
+/// Test get_recursive_transaction_results (called by get_final_transaction_result)
+/// only returns non-refund receipts
+#[test]
+fn test_refunds_not_in_receipts() {
+    test_with_client!(test_utils::NodeType::Validator, client, async move {
+        let block_hash = client.block(BlockReference::latest()).await.unwrap().header.hash;
+        let signer = InMemorySigner::from_seed("test1".parse().unwrap(), KeyType::ED25519, "test1");
+        let tx = SignedTransaction::send_money(
+            1,
+            "test1".parse().unwrap(),
+            "test2".parse().unwrap(),
+            &signer,
+            100,
+            block_hash,
+        );
+        let bytes = tx.try_to_vec().unwrap();
+        client.broadcast_tx_commit(to_base64(&bytes)).await.unwrap();
+        for _ in 1..10 {
```

Comment:
If I read this correctly, this loops always runs for the ten iterations, we never break from it, right? So this seems to be equivalent to just waiting for 100 millis.

I think what we want here is to loop until we do get a response, and then break. 

And we should assert that we actually did get a non-empty response in the end.

Reply:
updated to break when we get a response and asserting we get a non-empty response at the end